pkg/servicemesh/federation/status/manager_test.go

```diff
@@ -795,21 +757,20 @@ func TestStatusManager(t *testing.T) {
 			}
 			for index, f := range tc.events {
 				t.Logf("processing event %d", index)
-				f(handler)
-				verifyPeerStatus(t, cs, tc.mesh, &tc.status[index].peer, tc.assertions[index])
-				verifyExportStatus(t, cs, tc.mesh, &tc.status[index].exports)
-				verifyImportStatus(t, cs, tc.mesh, &tc.status[index].imports)
+				f.event(handler)
+				verifyPeerStatus(t, rm, tc.mesh, &f.expectedPeerStatus, f.peerStatusAssertion)
+				verifyExportStatus(t, rm, tc.mesh, &f.expectedExportStatus)
+				verifyImportStatus(t, rm, tc.mesh, &f.expectedImportStatus)
 			}
 		})
 	}
 }
 
-func verifyPeerStatus(t *testing.T, cs maistraclient.Interface, name types.NamespacedName, expected *v1.ServiceMeshPeerStatus,
-	assert func(*testing.T, *v1.ServiceMeshPeerStatus) error,
-) {
+func verifyPeerStatus(t *testing.T, rm common.ResourceManager, name types.NamespacedName, expected *v1.ServiceMeshPeerStatus, assert func(*testing.T,
+	*v1.ServiceMeshPeerStatus) error) {
 	t.Helper()
 	tryMultipleTimes(t, func() error {
-		actual, err := cs.FederationV1().ServiceMeshPeers(name.Namespace).Get(context.TODO(), name.Name, metav1.GetOptions{})
+		actual, err := rm.PeerInformer().Lister().ServiceMeshPeers(name.Namespace).Get(name.Name)
```

Comment:
I didn't know only the Lister goes through the cache. Will try to remember this one